### PR TITLE
Remove textures code

### PR DESCRIPTION
**Pull request by IagoLast**
#49 Since textured markers are buggy in tangram, it makes no sense to have code for that in our
library.

### CODE REVIEW

**Comment by rochoa** [comment]
I'm wondering if we should reconsider this, let me explain: what is buggy is using `marker-file` in combination with `marker-fill`, but using `marker-file` alone works pretty much OK. Could we detect that and only reject to render when both are applied? OTOH, does the default `maker-fill` color prevents us to take that approach?

**Comment by IagoLast** [comment]
@rochoa good catch!

I´ve been doing some research and as long as I know tangram-reference doesn´t have support for "conditional rules" .

This means we should check this type of issues at tangram-carto level, something like:

```javascript
if ((haveColor(point) && haveTexture(points)) {
  throw new Error('tangram rendering don\'t support colored markers');
}
```

this kind of things can be considered as "polluting" tangram-carto with logic but IMO as long as we keep them under control they can give us lot of flexibility

**Comment by rochoa** [comment]
If you agree, I would like to see that proposal.

**Comment by IagoLast** [comment]
The solution (maybe is not very elegant) is so simple that speaks for itself, see the [commit](https://github.com/CartoDB/tangram-carto/pull/51/commits/762d145cdda8e8e45d43bc263164acf450f16fa7)


### Test1: marker file with no color

builder:

screen shot 2017-08-04 at 10 09 03

Tangram result:

screen shot 2017-08-04 at 10 09 17


### Test2 marker file with color

builder:

screen shot 2017-08-04 at 10 10 45

tangram result:

screen shot 2017-08-04 at 10 11 18


Markers with no files work as expected:

screen shot 2017-08-04 at 10 12 20


Notice that markers are loosing resolution and color (maybe because of tangram, im not sure).

File: src/validator.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+function validateCSS(ccss) {
+  _validateNotFillFile(ccss);
+}
+
+/**
+ * Throw an exception if the ccss has colored file makers.
+ * Tangram cannot handle painting a image causing a bad result.
+ */
+function _validateNotFillFile(ccss) {
+  if ((ccss.indexOf('marker-fill') >= 0) && (ccss.indexOf('marker-file') >= 0)) {
```
Comment (alonsogarciapablo) [inline src/validator.js]: Using `indexOf` might give us some false positives. For example, when using the following CartoCSS (with conditionals):

```
#layer {
  marker-width: 7;
  marker-fill: #FFB927;
  marker-fill-opacity: 0.9;
  marker-allow-overlap: true;
  marker-line-width: 1;
  marker-line-color: #FFF;
  marker-line-opacity: 1;
}

#layer[name='Madrid'] {
  marker-width: 47;
  marker-file: url('https://s3.amazonaws.com/com.cartodb.users-assets.production/maki-icons/triangle-stroked-18.svg');
  marker-fill-opacity: 0.9;
  marker-allow-overlap: true;
  marker-line-width: 1;
  marker-line-color: #FFF;
  marker-line-opacity: 1;
}
```

File: src/validator.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    throw new Error('tangram don\'t support colored markers: marker-file and marker-fill properties are not compatible');
```
Comment (alonsogarciapablo) [inline src/validator.js]: I think "tangram *doesn't* support..." would ready better.

File: test/validator_spec.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/* globals describe, it */
+import Utils from './utils/utils.js';
+import Validator from '../src/validator.js';
+import chai from 'chai';
+const expect = chai.expect;
+
+describe('Validator', () => {
+  describe('.validateCCSS', () => {
+    it('Should throw an Error when valid ccss with incompatible rules is given', () => {
```
Comment (alonsogarciapablo) [inline test/validator_spec.js]: Perhaps worth adding a test that doesn't throw an error.

File: test/validator_spec.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      let invalidCCSS = `
+        #layer {
+          marker-file: url('https://s3.amazonaws.com/com.cartodb.users-assets.production/maki-icons/square-18.svg');
+          marker-line-color: #FFF;
+          marker-line-opacity: 1;
+          marker-comp-op: overlay;
+          marker-fill: #ccc;
+        }`;
+      expect(Utils.runFn(Validator.validateCSS, invalidCCSS)).to.throw();
```
Comment (alonsogarciapablo) [inline test/validator_spec.js]: We could just declare de function here and also use `to.throwError` which is a bit more explicit:

```
expect(function () {
  Validator.validateCSS(invalidCCS);
}).to.throwError("tangram doesn''t support...");

```

**Comment by alonsogarciapablo** [comment]
>  using marker-file alone works pretty much OK

From what I can see, BUILDER will always add both `marker-fill` and `marker-file` when an image is selected in the fill popup.

**Comment by alonsogarciapablo** [comment]
I asked @makella (thanks!) about this and it will be unlikely that someone will go edit the CartoCSS and remove the `marker-fill` after they have chosen an icon... I suggest we go with the solution that @IagoLast suggested initially (removing `marker-file` from https://github.com/CartoDB/tangram-reference).

**Comment by IagoLast** [comment]
ummm, Pablo you are right, there are some conditional css examples that cannot be cached by the `validator`

**Comment by alonsogarciapablo** [comment]
@IagoLast I guess we can close this PR and remove `marker-file` from tangram-reference (https://github.com/CartoDB/tangram-reference/blob/master/1.0.0/reference.json#L111-L118). Agree?

**Comment by IagoLast** [comment]
@alonsogarciapablo nope, due a bug the code for textures must be removed or tangram-carto will crash when we remove "marker-file" from the reference. 

We can update this PR or create a new one

**Comment by alonsogarciapablo** [comment]
Ok. I'm missing some context then. Feel free to do whatever it's easier for you then. Thanks!

**Comment by IagoLast** [comment]
Will create a new PR & branch

**Pull request closed without merging**